MathModule.lua

```diff
@@ -221,28 +220,10 @@ function Number:GetAbbSuffix()
 	local suffixIndex = math.floor(second/3)
 	local str = math.floor(first * 10)/10
 
-	if suffixIndex > 0 then
-		str ..= " " .. full_names[suffixIndex] or "e+"..second
-	end
-
-	return str
-end
-
-function Number:GetSuffix()
-	local numbers =  self.val:split(',')
-	local first, second = numbers[1], numbers[2]
-
-	first = tonumber(first)
-	second = tonumber(second)
-
-	local secondRemainder = second % 3
-	first *= 10^secondRemainder
-
-	local suffixIndex = math.floor(second/3)
-	local str = math.floor(first * 10)/10
+	local suffix = if abbreviation then suffixes[suffixIndex] else  " " .. (full_names[suffixIndex] or "")

```

Comment:
good step at tackling edge case, but still not completely resolved.
take a look at this scenario:
```lua
local full_names = {"A", "B", "C"}
local abbreviation = false
local suffixIndex = 4 -- this index is out of bounds

local suffix = if abbreviation then suffixes[suffixIndex] else " " .. (full_names[suffixIndex] or "")
-- at this point, suffix = " "

if suffixIndex > 0 then
 str ..= suffix or "e+"..second
end

-- str now equals "{number} " instead of "{number}e+{second}"
```

Reply:
My bad, was moving around code and forgot to update to return nil